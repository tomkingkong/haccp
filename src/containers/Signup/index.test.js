import React from 'react';
import { shallow } from 'enzyme';

import { Signup } from '.';

describe('Signup Component', () => {

  it('should match snapshot', () => {
    const wrapper = shallow(<Signup />);
    expect(wrapper).toMatchSnapshot();
	});

	it('should have default states', () => {
		const wrapper = shallow(<Signup />);
		const expected = {
			password: '',
			email: ''
		}
    expect(wrapper.state()).toEqual(expected);
	});
	
	it('should handle a change', () => {
		const wrapper = shallow(<Signup />);
		const mockEvent = {
			target: {
				name: 'email',
				value: 'ben@gmail.com'
			}
		};
		expect(wrapper.state('email')).toEqual('');
		wrapper.instance().handleChange(mockEvent);
		expect(wrapper.state('email')).toEqual('ben@gmail.com');
	});

	it('should handle submit', async () => {
		window.fetch = jest.fn().mockImplementation(() => Promise.resolve({id:1}));
		const mockFn = jest.fn();
		const mockHistory = {
			push: mockFn
		};
		const wrapper = shallow(<Signup 
			history={mockHistory} />);

		const mockEvent = {
			preventDefault: jest.fn()
		};
		await wrapper.instance().handleSubmit(mockEvent);
		expect(mockFn).toHaveBeenCalled();
		expect(mockFn).toHaveBeenCalledWith('/companyinfo');
	});

	describe('goToLogin', () => {
		it('should send the user to login page', () => {
			const mockFn = jest.fn();
			const mockHistory = {
				push: mockFn
			};
			const wrapper = shallow(<Signup 
				history={mockHistory}/>);
			wrapper.instance().goToLogin();
			expect(mockFn).toHaveBeenCalled();
			expect(mockFn).toHaveBeenCalledWith('/login');
		});
	});
});