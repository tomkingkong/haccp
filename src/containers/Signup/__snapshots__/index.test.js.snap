// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component should match snapshot 1`] = `
<div
  className="signup"
>
  <h2>
    Signup
  </h2>
  <form
    onSubmit={[Function]}
  >
    <input
      name="email"
      onChange={[Function]}
      placeholder="Email"
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="Password"
      value=""
    />
    <button>
      Sign Up
    </button>
  </form>
  <p>
    Already have an account: 
  </p>
  <button
    onClick={[Function]}
  >
    Log In
  </button>
</div>
`;
